libhb/enc_qsv.c

```diff
@@ -1404,7 +1404,7 @@ int encqsvInit(hb_work_object_t *w, hb_job_t *job)
     }
 
     // AsyncDepth has now been set and/or modified by Media SDK
-    pv->max_async_depth = videoParam.AsyncDepth;
+    pv->max_async_depth = videoParam.AsyncDepth > 0 ? videoParam.AsyncDepth : AV_QSV_ASYNC_DEPTH_DEFAULT;
```

Comment:
AsyncDepth is unsigned, so you could get rid of > 0; though the intent might be clearer if you keep it. Either are fine by me. I would add a comment like:

// AsyncDepth has now been set and/or modified by Media SDK
// fall back to default if zero (invalid value)

Thanks!